File: tmt-translator-extension/setup/generate-icons.py
#!/usr/bin/env python3
"""
Generate placeholder icons for the TMT Translator Extension
Run this script once to create the required icon files
"""

import os
from PIL import Image, ImageDraw

def create_icon(size, filename):
    """Create a simple circular icon with the given size"""
    # Create a new image with green background
    img = Image.new('RGB', (size, size), color='#4CAF50')
    draw = ImageDraw.Draw(img)
    
    # Add a white circle border
    border = int(size * 0.1)
    draw.ellipse(
        [border, border, size - border, size - border],
        outline='white',
        width=2
    )
    
    # Add a "G" for Google/Global in the center
    try:
        # Try to use a built-in font, if available
        draw.text(
            (size // 2, size // 2),
            "🌐",
            fill='white',
            anchor='mm'
        )
    except:
        # Fallback: just save without text if font fails
        pass
    
    # Save the image
    img.save(filename)
    print(f"✓ Created {filename} ({size}x{size})")

def main():
    """Generate all required icons"""
    asset_dir = os.path.join(os.path.dirname(__file__), '..', 'src', 'assets')
    
    # Create assets directory if it doesn't exist
    os.makedirs(asset_dir, exist_ok=True)
    
    # Create icons for each required size
    icons = [
        (16, os.path.join(asset_dir, 'icon-16.png')),
        (48, os.path.join(asset_dir, 'icon-48.png')),
        (128, os.path.join(asset_dir, 'icon-128.png')),
    ]
    
    print("Generating extension icons...")
    print("-" * 40)
    
    try:
        for size, filepath in icons:
            create_icon(size, filepath)
        
        print("-" * 40)
        print("✓ All icons generated successfully!")
        print(f"\nIcons saved to: {asset_dir}/")
        print("\nYou can now load the extension in Chrome:")
        print("  1. Go to chrome://extensions/")
        print("  2. Enable Developer mode")
        print("  3. Click 'Load unpacked'")
        print("  4. Select the tmt-translator-extension folder")
        
    except ImportError:
        print("ERROR: Pillow library not found!")
        print("\nTo fix this, install Pillow:")
        print("  pip install pillow")
        print("\nOr use the alternative method:")
        print("  1. Download icons from: https://www.favicon-generator.org/")
        print("  2. Create 16x16, 48x48, and 128x128 PNG files")
        print("  3. Place them in: src/assets/")
        return False
    
    return True

if __name__ == '__main__':
    success = main()
    exit(0 if success else 1)

File: tmt-translator-extension/setup/generate-icons.js
#!/usr/bin/env node
/**
 * Generate placeholder icons for the TMT Translator Extension
 * Node.js version using canvas library
 * 
 * Installation: npm install canvas
 */

const fs = require('fs');
const path = require('path');

// Try to use canvas, fall back if not available
let Canvas;
try {
  Canvas = require('canvas').Canvas;
} catch (e) {
  console.error('ERROR: canvas library not found!');
  console.error('\nTo use this script, install the canvas library:');
  console.error('  npm install canvas');
  console.error('\nOr use the Python version instead:');
  console.error('  python setup/generate-icons.py');
  process.exit(1);
}

function createIcon(size, filepath) {
  const canvas = new Canvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Draw green background
  ctx.fillStyle = '#4CAF50';
  ctx.fillRect(0, 0, size, size);
  
  // Draw white circle
  ctx.strokeStyle = 'white';
  ctx.lineWidth = Math.max(1, Math.floor(size * 0.1));
  ctx.beginPath();
  ctx.arc(size / 2, size / 2, (size - ctx.lineWidth) / 2, 0, Math.PI * 2);
  ctx.stroke();
  
  // Draw globe emoji or text
  ctx.fillStyle = 'white';
  ctx.font = `bold ${Math.floor(size * 0.6)}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  try {
    ctx.fillText('🌐', size / 2, size / 2);
  } catch (e) {
    ctx.fillText('T', size / 2, size / 2);
  }
  
  // Save to file
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(filepath, buffer);
  console.log(`✓ Created ${path.basename(filepath)} (${size}x${size})`);
}

function main() {
  const assetDir = path.join(__dirname, '..', 'src', 'assets');
  
  // Create assets directory if it doesn't exist
  if (!fs.existsSync(assetDir)) {
    fs.mkdirSync(assetDir, { recursive: true });
  }
  
  console.log('Generating extension icons...');
  console.log('-'.repeat(40));
  
  const icons = [
    { size: 16, file: 'icon-16.png' },
    { size: 48, file: 'icon-48.png' },
    { size: 128, file: 'icon-128.png' },
  ];
  
  try {
    icons.forEach(({ size, file }) => {
      const filepath = path.join(assetDir, file);
      createIcon(size, filepath);
    });
    
    console.log('-'.repeat(40));
    console.log('✓ All icons generated successfully!\n');
    console.log(`Icons saved to: ${assetDir}/`);
    console.log('\nYou can now load the extension in Chrome:');
    console.log('  1. Go to chrome://extensions/');
    console.log('  2. Enable Developer mode');
    console.log('  3. Click "Load unpacked"');
    console.log('  4. Select the tmt-translator-extension folder');
    
  } catch (error) {
    console.error('ERROR:', error.message);
    process.exit(1);
  }
}

main();
